build/common.props

```diff
@@ -16,5 +16,29 @@
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <WarningsAsErrors />
     <SharedVersionOutputDirectory>$(MSBuildThisFileDirectory)\src\Shared</SharedVersionOutputDirectory>
+
+    <!--
+      Generate assembly info so that IntervalsVisibleTo
+      can be specified in ProjectFile, where we have access to
+      AssemblyNamePrefix
+    -->
+    <GenerateAssemblyInfo>true</GenerateAssemblyInfo>
+    
+    <!--
+      Disable generation of assembly properties where their values
+      differ for NuGet packages and assemblies. For these
+      properties are defined in AssemblyInfo files.
```

Comment:
I wonder: why not just remove all AssemblyInfo files and specify assembly properties such as names etc in the project files instead (if they differ from default/generated ones)?

Reply:
"specify assembly properties in the project files instead"

The way this works is that the things you specify in the project file are the properties of the NuGet package. And it can autogenerate the assembly properties to be the same, but if you want them to be different you have to use an AssemblyInfo.cs file.

Reply:
Sorry, I missed the "Nuget" part. 

Especially if shipping/publishing to multiple parties, coupling project files with NuGet packages is a debatable decision IMO. But again, clearly not this PR.